File: Project/python/citta.py
def main():
    num_citta = 0
    abitanti_totali = 0

    citta_max = ""
    abitanti_max = 0

    abitanti_5k = True
    madrid = False
    parigi = False

    tentativi = 0

    while tentativi < 10:
        citta = input("Inserisci la citta: ")

        if citta == "Oslo":
            break

        abitanti = int(input("Inserisci il numero di abitanti: "))

        abitanti_totali += abitanti
        num_citta += 1

        if abitanti > abitanti_max:
            abitanti_max = abitanti
            citta_max = citta

        if citta == "Madrid":
            madrid = True
        if citta == "Parigi":
            parigi = True

        if abitanti <= 5000:
            abitanti_5k = False

        tentativi += 1

    if num_citta > 0:
        abitanti_media = abitanti_totali / num_citta
        print(f"Media abitanti: {abitanti_media}")
        print(f"Città con più abitanti: {citta_max} con {abitanti_max} abitanti.")
        print(f"Città Madrid presente: {'Sì' if madrid else 'No'}")
        print(f"Città Parigi presente: {'Sì' if parigi else 'No'}")
        print(f"Tutte le città hanno più di 5000 abitanti: {'Sì' if abitanti_5k else 'No'}")
    else:
        print("Nessuna città inserita.")

if __name__ == "__main__":
    main()

File: Project/C & C++/citta.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
    int abitanti;
    string citta;

    int numCitta = 0;
    int abitantiTotali = 0; // Per calcolare la media

    string cittaMax;
    int abitantiMax = 0;

    bool abitanti5k = true;
    bool Madrid = false;
    bool Parigi = false;

    int tentativi = 0;

    while (tentativi < 10) {
        cout << "Inserisci la citta: ";
        cin >> citta;

        if (citta == "Olso") {
            break; // Esci se la città è "Olso"
        }

        cout << "Inserisci il numero di abitanti: ";
        cin >> abitanti;

        abitantiTotali += abitanti;
        numCitta++;

        if (abitanti > abitantiMax) {
            abitantiMax = abitanti;
            cittaMax = citta;
        }

        
        if (citta == "Madrid") {
            Madrid = true;
        }
        if (citta == "Parigi") {
            Parigi = true;
        }

        if (abitanti <= 5000) {
            abitanti5k = false;
        }

        tentativi++;
    }

    if (numCitta > 0) {
        double abitantiMedia = static_cast<double>(abitantiTotali) / numCitta;
        cout << "Media abitanti: " << abitantiMedia << endl;
        cout << "Città con più abitanti: " << cittaMax << " con " << abitantiMax << " abitanti." << endl;
        cout << "Città Madrid presente: " << (Madrid ? "Sì" : "No") << endl;
        cout << "Città Parigi presente: " << (Parigi ? "Sì" : "No") << endl;
        cout << "Tutte le città hanno più di 5000 abitanti: " << (abitanti5k ? "Sì" : "No") << endl;
    } else {
        cout << "Nessuna città inserita." << endl;
    }

    return 0;
}